packages/@dcl/inspector/src/components/Controls/Controls.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from 'react'
+import { MdOutlineDriveFileRenameOutline } from 'react-icons/md'
+import { AiFillFileAdd, AiFillDelete } from 'react-icons/ai'
+import { RxListBullet } from 'react-icons/rx'
+
+import './Controls.css'
+
+interface ControlsProps {
+  enableAdd?: boolean
+  enableEdit?: boolean
+  enableRemove?: boolean
+  handleEdit: (e: React.MouseEvent) => void
+  handleAdd: (e: React.MouseEvent) => void
+  handleRemove: (e: React.MouseEvent) => void
+}
+
+const cancelingKeys = new Set(['Escape', 'Tab'])
+
+const getActiveClass = (active: boolean) => active ? 'active' : ''
+
+const Controls = ({
+  enableAdd = true,
+  enableEdit = true,
+  enableRemove = true,
+  handleEdit,
+  handleAdd,
+  handleRemove
+}: ControlsProps) => {
+  const [active, setActive] = useState(false)
+
+  const onCancel = (e: Event) => {
+    e.stopPropagation()
+    setActive(false)
+  }
+
+  const onKeyUp = (e: KeyboardEvent) => {
+    if (cancelingKeys.has(e.key)) onCancel(e)
+  }
+
+  useEffect(() => {
+    if (active) {
+      document.body.addEventListener('click', onCancel)
+      document.body.addEventListener('keyup', onKeyUp)
+    } else {
+      document.body.removeEventListener('click', onCancel)
+      document.body.removeEventListener('keyup', onKeyUp)
```

Comment:
if the component is unmounted while it was active, these lines will not execute. We should also remove the listeners when the component unmounts to avoid calling these handlers which would fail when the component is not mounted:

```ts
useEffect(() => {
  return () => {
    document.body.removeEventListener('click', onCancel)
    document.body.removeEventListener('keyup', onKeyUp)
  }
}, [onCancel, onKeyUp])
```

And also lets wrap all the hanlders (onCancel,  onKeyUp, handleClick, handleAction) with `useCallback` to prevent extra renders and effects.